enhancements/authentication/pod-security-admission-enforcement.md

```diff
@@ -0,0 +1,486 @@
+Without PSA enforcement it sets the `PodSecurity` configuration of the kube-apiserver based on the `OpenShiftPodSecurityAdmission` `FeatureGate`.
+
+The Config Observer must watch:
+- the `status.enforcementMode`, 
+- the `FeatureGate` `OpenShiftPodSecurityAdmission` and
+- check whether all Namespaces that are managed by the `PodSecurityAdmissionLabelSynchronizationController` are labeled.
+
+Only when all conditions are in a `Restricted` mode, will the Config Observer change the kube-apiserver's `PodSecurity` configuration.
+This means:
+- `status.enforcementMode = Restricted`
+- `FeatureGate` `OpenShiftPodSecurityAdmission` is enabled
+- all Namespaces that are managed by the `PodSecurityAdmissionLabelSynchronizationController` are labeled.
+
+The `API` and `FeatureGate` state needs to be watched continuously.
+If any of those two restrictive values change, the `PodSecurity` configuration must stop enforcing `Restricted` on Namespaces without `pod-security.kubernetes.io/enforce` label.
+
+#### PodSecurityAdmissionLabelSynchronizationController
+
+The [PodSecurityAdmissionLabelSynchronizationController (PSA label syncer)](https://github.com/openshift/cluster-policy-controller/blob/master/pkg/psalabelsyncer/podsecurity_label_sync_controller.go) labels all the Namespaces it manages.
+Without PSA enforcement it sets the `pod-security.kubernetes.io/warn` and `pod-security.kubernetes.io/audit` labels on managed Namespaces.
+
+Namespaces that are **managed** by the `PodSecurityAdmissionLabelSynchronizationController` are Namespaces that:
+
+- are not named `kube-node-lease`, `kube-system`, `kube-public`, `default` or `openshift` and
+- are not  prefixed with `openshift-` and
+- have no `security.openshift.io/scc.podSecurityLabelSync=false` label set and
+- at least one PSA label (including `pod-security.kubernetes.io/enforce`) isn't set by the user or
+- if the user sets all PSA labels, it also has set the `security.openshift.io/scc.podSecurityLabelSync=true` label.
+
+The PSA label syncer must watch:
+- the `status.enforcementMode`and 
+- the `FeatureGate` `OpenShiftPodSecurityAdmission`.
+
+Only when all conditions are in a `Restricted` mode, the PSA label syncer will label managed Namespaces with `pod-security.kubernetes.io/enforce` as evaluated in the `security.openshift.io/MinimallySufficientPodSecurityStandard` annotation.
+This means:
+- `status.enforcementMode = Restricted` and
+- `FeatureGate` `OpenShiftPodSecurityAdmission` is enabled.
+
+This state needs to be watched continuously.
+If any of the values changes, the `PodSecurity` configuration must stop enforcing `Restricted` on Namespaces without `pod-security.kubernetes.io/enforce` label.
+
+Because the ability to set `pod-security.kubernetes.io/enforce` is introduced, the ability to remove that label must exist in the release before.
+Otherwise, the cluster will be unable to revert to its previous state.
+
+## Open Questions
+
+## Rerun the evaluation before enforcing
```

Comment:
It sounds reasonable to add alerts in release n and run the actual final evaluation right before potentially enforcing as between release n and release n+1 could be a gap of months.